backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs

```diff
@@ -1,29 +1,59 @@
 ﻿using AutoMapper;
 using EasyMeets.Core.BLL.Interfaces;
-using EasyMeets.Core.DAL.Context;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using EasyMeets.Core.Common.DTO.Availability;
+using EasyMeets.Core.DAL.Context;
+using Microsoft.EntityFrameworkCore;
 using EasyMeets.Core.Common.DTO.Availability.NewAvailability;
 using EasyMeets.Core.DAL.Entities;
+using EasyMeets.Core.Common.Enums; 
 
 namespace EasyMeets.Core.BLL.Services
 {
     public class AvailabilityService : BaseService, IAvailabilityService
     {
-        public AvailabilityService(EasyMeetsCoreContext context, IMapper mapper) : base(context, mapper)
+        public AvailabilityService(EasyMeetsCoreContext context, IMapper mapper) : base(context, mapper) { }
+        public async Task<ICollection<AvailabilitySlotsGroupByTeamsDto>> GetAllAvailabilitySlotsGroupByTeamsAsync(int id)
         {
+            var teamsWithSlots = await _context.Teams
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Members)
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Location)
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Author)
+                .Where(x => x.AvailabilitySlots.Any(x => x.Type == SlotType.Team))
+                .Where(x => x.AvailabilitySlots.Any(x => x.Members.Any(x => x.UserId == id)))
+                .ToListAsync();
```

Comment:
This is not a good idea to load all data and only after that map it to a DTO
It would be better to do a projection here and load only necessary fields